### ISSUE TITLE
RFC: first release

### ISSUE THREAD

**Issue by alokat** [andres-erbsen/dename#40]
Hi, 

I wonder if it is time for a first release. This would make it easier to port dename on platforms, like *BSD, because of the ports tree. The version could start with 0.1, in order to show it is not ready for production yet. The ports tree downloads the tar.gz or zip archive and build the project. It does
not make a checkout of a given branch. That's were the release comes in. If a new one was created it can be downloaded from github as such an archive.

**Comment by andres-erbsen** [comment]
I agree that it is time to publish something with a version number after #42. I will probably tag v0.3 after #42 gets closed (and hopefully sign it using a gpg key key that is still valid, working on that...). v0.3 because .1 and .2 have already existed, in a repo whose history is not mirrored here but which was published.

In fact, I am not foreseeing any significant changes to the server or the client that would not mean a total rewrite:

- There will probably be bugs, and they will hopefully get fixed.
- A less restrictive invite system that would still not let one person hoard thousands of names would be very welcome, and could probably be accomplished somehow (phone numbers? bitcoin?), but it is not a priority for me and will this not get done any time soon.
- In an ideal world, this entire thing would be rewritten to use the Stellar Consensus Protocol instead of hand-picked leaders. But this would be a huge change AND the [SCP github page](https://github.com/stellar/stellar-core) says it is pre-beta, so I am not in a hurry to go for that just now.

**Comment by alokat** [comment]
Sounds like there is a kind of roadmap needed. So we can focus on changes that are more important as
others.
Personally, I prefer "evolution not revolution". Cause small steps to reach a greater goal is much easier to handle then one single step. If it's for sure, that SCP will be used in the future I'm sure there are things that can already be prepared now to make the final change not that hard.

**Comment by andres-erbsen** [comment]
https://github.com/andres-erbsen/dename/releases/tag/0.3 is a thing now.

    oldie@ashryn ..go/src/github.com/andres-erbsen/dename (git)-[master] % git describe  --tags
    0.3
    oldie@ashryn ..go/src/github.com/andres-erbsen/dename (git)-[master] % git log | head -1         
    commit bb79881dfbd81ebd43c8197518d0e5974a94c501

    pub   rsa4096/5200E3F8 2015-07-16 [expires: 2017-07-15]
          Key fingerprint = D215 2255 DFF3 8C0B 549D  3DEF 5103 9942 5200 E3F8
    uid       [  full  ] Andres Erbsen (dename code signing key) <andreser@mit.edu>
    sig!3        5200E3F8 2015-07-16  Andres Erbsen (dename code signing key) <andreser@mit.edu>
    sig!         D6DBA0CF 2015-07-16  Andres Erbsen <andreser@mit.edu>

I still haven't figured out how to upload a signed tarball to github releases page without having it create a new release...

**Comment by alokat** [comment]
Great, I will a hack a port for OpenBSD during the next days.

**Comment by alokat** [comment]
I think it's better to name the release `dename-<version number>` instead of just `<version number>`.

If it is just a number the download looks like `0.3.{tar.gz|zip}`. This is a little bit confusing, if you
download it using bsd ports. There is just a `0.3` directory, but which tool does it contain? :)
```
$ git tag dename-0.3 0.3
$ git tag -d 0.3
$ git push origin :refs/tags/0.3
$ git push --tags
```
This should do the trick.
Now the extracted dir is `dename-0.3` which is much more significant.

**Comment by alokat** [comment]
Maybe this is not need yet. :)
I got a little bit more involved with BSD ports and gh projects and they support it already by using
special variables - GH_.

I will report more, if my port was commited to the tree.

**Comment by alokat** [comment]
Closed ....
